ngraph/frontend/onnx_import/src/editor/editor.cpp

```diff
@@ -369,3 +403,118 @@ void onnx_import::ONNXModelEditor::set_input_values(
         modify_initializer(*onnx_initializer, name, values, onnx_input);
     }
 }
+
+void onnx_import::ONNXModelEditor::ONNXModelEditor::replace_nodes(
+    std::vector<std::vector<int>> node_indexes, Operator node_generator)
+{
+    std::string new_op_name = "custom_op_" + std::to_string(m_custom_op_ID++);
+
+    shift_node_indexes(node_indexes);
+
+    for (auto node_vector : node_indexes)
+    {
+        auto new_end = std::unique(node_vector.begin(), node_vector.end());
+        node_vector.erase(new_end, node_vector.end());
+        replace_nodes(node_vector, node_generator, new_op_name);
+    }
+    onnx_import::register_operator(new_op_name, 1, "org.openvinotoolkit.editor", node_generator);
+}
+
+void onnx_import::ONNXModelEditor::ONNXModelEditor::replace_nodes(
+    std::vector<int> node_indexes, onnx_import::Operator node_generator, std::string new_op_name)
+{
+    std::vector<std::string> inputs;
+    std::vector<std::string> outputs;
+
+    auto nodes = model().graph().node();
+
+    // 1. create a list of inputs and outputs of new custom op.
+    for (auto node_index : node_indexes)
+    {
+        inputs.insert(
+            inputs.end(), nodes[node_index].input().begin(), nodes[node_index].input().end());
```

Comment:
Potential segfault if `node_index > nodes.size()`. The same in the instruction below.